HopsanCore/src/ComponentSystem.cpp

```diff
@@ -701,66 +701,67 @@ bool ComponentSystem::sortComponentVector(std::vector<Component*> &rComponentVec
     }
 #else
     bool didSomething = true;
-    while(didSomething)
-    {
+    while(didSomething) {
         didSomething = false;
 
         // Loop through the unsorted component vector
-        for(size_t c=0; c<rComponentVector.size(); ++c)
-        {
+        for(size_t c=0; c<rComponentVector.size(); ++c) {
             // A pointer to an unsorted component
             Component* pUnsrtComp = rComponentVector[c];
             //Ignore components that are already added to the new vector
-            if(!vectorContains<Component*>(newComponentVector, pUnsrtComp))
-            {
+            if(!vectorContains<Component*>(newComponentVector, pUnsrtComp)) {
                 bool readyToAdd=true;
                 std::vector<Port*> portVector = pUnsrtComp->getPortPtrVector();
                 // Ask each read port for its node, then ask the node for its write port component
-                for(size_t p=0; p<portVector.size(); ++p)
-                {
+                for(size_t p=0; p<portVector.size(); ++p) {
                     // Take the port pointer (To make code easier to read)
                     Port *pPort = portVector[p];
-                    Component* pRequiredComponent=0;
+                    std::vector<Component*> pRequiredComponents;
 
                     SortHintEnumT sortHint = pPort->getSortHint();
                     if ((pUnsrtComp->getTypeName() == HOPSAN_BUILTIN_TYPENAME_SUBSYSTEM) ||
-                        (pUnsrtComp->getTypeName() == HOPSAN_BUILTIN_TYPENAME_CONDITIONALSUBSYSTEM))
-                    {
+                        (pUnsrtComp->getTypeName() == HOPSAN_BUILTIN_TYPENAME_CONDITIONALSUBSYSTEM)) {
                         sortHint = pPort->getInternalSortHint();
                     }
 
-                    if ( (sortHint == Destination) && pPort->isConnected() )
-                    {
-                        Port *pSourcePort = pPort->getNodePtr()->getSortOrderSourcePort();
-                        if (pSourcePort && pSourcePort->getComponent())
-                        {
-                            pRequiredComponent = pSourcePort->getComponent();
+                    if ( (sortHint == Destination) && pPort->isConnected() ) {
+                        std::vector<Port *> sourcePorts;
+                        if(pPort->isMultiPort()) {
```

Comment:
I think that the for loop would be enough, and not need to check if multiport or not. The getNumPorts returns 1 for ordinary ports. But no need to change, I think this is more clear.

Reply:
True, it will work without the check. It is a bit ugly though that `getNodePtr(const size_t subPortIdx)` will ignore the index argument for non-multiports. This means that `getNodePtr(42) = getNodePtr(0) = getNodePtr()`. But perhaps this was the intended behaviour?

Reply:
yes it was intended by its not very nice behavior, should redesign at some point